docs/data/data-grid/list-view/ListView.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  GridRenderCellParams,
+  GridListColDef,
+  GridColDef,
+} from '@mui/x-data-grid-pro';
+import Box from '@mui/material/Box';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import Avatar from '@mui/material/Avatar';
+import Typography from '@mui/material/Typography';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import IconButton from '@mui/material/IconButton';
+import MessageIcon from '@mui/icons-material/Message';
+
+function ListViewCell(params: GridRenderCellParams) {
+  return (
+    <Stack
+      direction="row"
+      sx={{
+        alignItems: 'center',
+        height: '100%',
+        gap: 2,
+      }}
+    >
+      <Avatar sx={{ width: 32, height: 32, backgroundColor: params.row.avatar }} />
+      <Stack>
+        <Typography variant="body2" fontWeight={500}>
+          {params.row.name}
+        </Typography>
+        <Typography variant="body2" color="text.secondary">
+          {params.row.position}
+        </Typography>
+      </Stack>
+    </Stack>
+  );
+}
+
+const listColDef: GridListColDef = {
+  field: 'listColumn',
+  renderCell: ListViewCell,
+};
+
+const VISIBLE_FIELDS = ['avatar', 'name', 'position'];
+
+export default function ListView() {
+  const [isListView, setIsListView] = React.useState(true);
+
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    rowLength: 20,
+    visibleFields: VISIBLE_FIELDS,
+  });
+
+  const columns: GridColDef[] = React.useMemo(() => {
+    return [
+      ...data.columns,
+      {
+        type: 'actions',
```

Comment:
it is strange that using list view and list column you get actions still from the regular columns

Reply:
@cherniavskii thoughts on this? I wonder if `listColumn` should become `listColumns`, and we require users to specify actions for list view and regular view separately.

Reply:
I thought about this too. It totally makes sense for the actions column use case, especially for the `ListViewAdvanced` demo.
But should we limit the number of items in `listColumns` to 2? And how do we distribute the width between them? Any ideas?

Reply:
We decided not to persist the actions column, and make actions part of the list view cell instead.
This way, the list view cell always takes 100% of the width.
Done in [`034c355` (#14393)](https://github.com/mui/mui-x/pull/14393/commits/034c355ac1ef7a9789c74a5d60d1918f5fb604da)